### ISSUE TITLE
Relationship Filtering not working as expected

### ISSUE THREAD

**Issue by denolfe** [payloadcms/plugin-relationship-object-ids#1]
Filtering documents via a relationship does not return a proper result. No results are returned.

**Comment by denolfe** [comment]
Issue Recreated.

<img width="1163" alt="CleanShot 2023-06-15 at 12 13 23@2x" src="https://github.com/payloadcms/plugin-relationship-object-ids/assets/65888/bd28b163-28b7-4ae8-9c17-9fe258220d07">

API Data returns IDs properly
```
{
  "docs": [
    {
      "id": "648b369c2d7402142278d62b",
      "title": "Without Plugin",
      "author": {
        "id": "648b349ec64ac8c2a8c7a3b8",
        "name": "asdf",
        "email": "dev@payloadcms.com",
        "createdAt": "2023-06-15T15:56:14.841Z",
        "updatedAt": "2023-06-15T15:56:14.917Z"
      },
      "status": "draft",
      "createdAt": "2023-06-15T16:04:44.515Z",
      "updatedAt": "2023-06-15T16:04:44.515Z"
    },
    {
      "id": "648b3656c64ac8c2a8c7a3ec",
      "title": "With Plugin",
      "author": {
        "id": "648b349ec64ac8c2a8c7a3b8",
        "name": "asdf",
        "email": "dev@payloadcms.com",
        "createdAt": "2023-06-15T15:56:14.841Z",
        "updatedAt": "2023-06-15T15:56:14.917Z"
      },
      "status": "draft",
      "createdAt": "2023-06-15T16:03:34.276Z",
      "updatedAt": "2023-06-15T16:13:08.237Z"
    }
  ],
  "totalDocs": 2,
  "limit": 10,
  "totalPages": 1,
  "page": 1,
  "pagingCounter": 1,
  "hasPrevPage": false,
  "hasNextPage": false,
  "prevPage": null,
  "nextPage": null
}
```


The values in the DB differ (as expected). We likely need to make changes to core to accommodate the differences during relationship filtering/querying.

![CleanShot 2023-06-15 at 12 15 36@2x](https://github.com/payloadcms/plugin-relationship-object-ids/assets/65888/e324036d-db73-484b-b0db-5b58154657b5)

**Comment by jmikrut** [comment]
Fixed in Payload 1.9.5
